crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -667,9 +669,6 @@ impl<'db> TypeInferenceBuilder<'db> {
         match definition.kind(self.db) {
             DefinitionKind::Function(function) => self.infer_function_deferred(function.node()),
             DefinitionKind::Class(class) => self.infer_class_deferred(class.node()),
-            DefinitionKind::AnnotatedAssignment(_annotated_assignment) => {
-                // TODO self.infer_annotated_assignment_deferred(annotated_assignment.node());
-            }
```

Comment:
TODO: add a comment on why annotated assignment are not deferred

Reply:
I'm curious to read this comment!

Could we add some tests that annotations on annotated assignments in stub files and when `from __future__ import annotations` is active, are in fact deferred? (That is, can contain forward references.)

Reply:
Added the test cases in `assignment/annotations.md`